### PR TITLE
Improve torch.linalg.qr

### PR DESCRIPTION
**Pull request by antocuni**
This is a follow up of PR #47764 to fix the remaining details.

### CODE REVIEW

**Comment by antocuni** [comment]
@mruberry as promised, here is a follow up PR to fix all the remaining details of PR #47764.
I think I addressed all of your comments apart from this which I didn't understand:
https://github.com/pytorch/pytorch/pull/47764#pullrequestreview-558754924

(testing are still running, I'll check their status tomorrow)

**Comment by antocuni** [comment]
@mruberry tests seem to be ok and the failure unrelated so it's ready for review

**Review by mruberry — APPROVED** [review]
Awesome!

This looks great. The remaining issue is pretty niche. Let's not worry about it for now. When we review the torch.linalg functions ahead of the 1.8 release we can see if there are additional tweaks, then.

**Comment by mruberry** [comment]
Docs failures actually are real, although they're very hard to parse:

```
Jan 05 15:57:22 docstring of torch.linalg.qr:29: WARNING: Explicit markup ends without a blank line; unexpected unindent.
```

Just ping me when this is ready to merge.

**Comment by antocuni** [comment]
> Just ping me when this is ready to merge.

fixed the issue, the docs jobs are now green. Other jobs are still in-progress though, I'll take care to ping you when they are green for real

**Comment by antocuni** [comment]
@mruberry all tests are green 👍
